src/wallet/api/wallet2_api.h

```diff
@@ -1144,6 +1154,519 @@ struct Wallet
+    * param: account_index - major index
+    * param: address_index - minor index
+    */
+    virtual void createOneOffSubaddress(std::uint32_t account_index, std::uint32_t address_index) = 0;
+    /**
+    * brief: getWalletState - get information about the wallet
+    *                         - is wallet file format deprecated
+    * return: WalletState object
+    */
+    virtual WalletState getWalletState() const = 0;
+    /**
+    * brief: hasUnknownKeyImages - check if any enotes have missing key images
+    * return: true if any stored enote is missing its key image, else false
+    */
+    virtual bool hasUnknownKeyImages() const = 0;
+    /**
+    * brief: rewrite - rewrite the wallet file for wallet update
+    * param: wallet_name - name of the wallet file (should exist)
+    * param: password - wallet password
+    * note: sets status error on fail
+    */
+    virtual void rewrite(const std::string &wallet_name, const std::string &password) = 0;
+    // QUESTION : Should we change this function from the current behavior in wallet2, so `wallet_name` is just the name of the new wallet instead of changing the `m_wallet_file` for the current wallet?
+    /**
+    * brief: writeWatchOnlyWallet -  create a new watch-only wallet file with view keys from current wallet
+    * param: wallet_name - name of the current wallet file
+    * param: password - password for new watch-only wallet
+    * outparam: new_keys_file_name - wallet_name + "-watchonly.keys"
+    * note: sets status error on fail
+    */
+    virtual void writeWatchOnlyWallet(const std::string &wallet_name, const std::string &password, std::string &new_keys_file_name) = 0;
+    /**
+    * brief: balancePerSubaddress - get balance for each subaddress in given account
+    * param: index_major - account index
+    * param: strict -
+    * return: [minor subaddress index: amount, ... ]
+    */
+    virtual std::map<std::uint32_t, std::uint64_t> balancePerSubaddress(std::uint32_t index_major, bool strict) const = 0;
+    /**
+    * brief: unlockedBalancePerSubaddress - get unlocked balance for each subaddresses in given account
+    * param: index_major - account index
+    * param: strict -
+    * return: [minor subaddress index: {amount, {blocks to unlock, time to unlock} }, ... ]
+    */
+    virtual std::map<std::uint32_t, std::pair<std::uint64_t, std::pair<std::uint64_t, std::uint64_t>>> unlockedBalancePerSubaddress(std::uint32_t index_major, bool strict) const = 0;
+    // QUESTION : Can anyone help with this comment? There are other ones I'm not satisfied with, so if you have a better suggestion I'll change them.
+    /**
+    * brief: isTransferUnlocked - check if transfer is unlocked
+    * param: unlock_time -
+    * param: block_height -
+    * return: true if transfer is unlocked, else false
+    */
+    virtual bool isTransferUnlocked(std::uint64_t unlock_time, std::uint64_t block_height) const = 0;
```

Comment:
Is this method lacking a parameter, the transfer in question?

Anyway, maybe a candidate to go because itt may be very simple to check yourself whether a transfer is unlocked or not yet, if the info given back about enotes is complete and contains the unlock height.

Reply:
It's the equivalent to [this](https://github.com/monero-project/monero/blob/master/src/wallet/wallet2.cpp#L7268), which is used by wallet-RPC and -CLI.

But I agree on the second part.